snuba/datasets/entities/discover.py

```diff
@@ -420,13 +423,39 @@ def selector_func(_query: Query) -> Tuple[str, List[str]]:
 
         def callback_func(results: List[Result[QueryResult]]) -> None:
             primary_result = results.pop(0)
+            primary_result_data = primary_result.result.result["data"]
 
             for result in results:
-                if result.result.result["data"] == primary_result.result.result["data"]:
+                result_data = result.result.result["data"]
+
+                if result_data == primary_result_data:
```

Comment:
Since these queries are executed kind of in parallel but while we are still ingesting data. It will be hard to see exactly the same results especially on long queries. I thinkn we will need something a bit more sophisticated to actually get meaningful results.

I think a better approach is to research some known similarity metrics to calculate on the two datasets. (Some examples are in this series https://towardsdatascience.com/a-practitioners-guide-to-similarity-scoring-part-1-29625b9ac481. But there are a lot more, we should spend some time finding a good one for our use case).

And we already did something similar when migrating to Snuba. https://blog.sentry.io/2019/07/25/field-guide-mitigating-risk-transitioning-databases/